python/tank_vendor/shotgun_deploy/descriptor.py

```diff
@@ -295,3 +295,63 @@ def find_latest_version(self, constraint_pattern=None):
         return latest
 
 
+###########################################################################
+# backward compatibility support - not for external use
+
+def get_legacy_cache_path(
```

Comment:
Hmm ok I see! Makes sense! I think it would still be good to keep this encapsulated inside `IODescriptor` - that's where all the low level functionality is being kept that handles the "transport" level functionality of the descriptors, e.g. where they are being stored on disk, how to update them, find them etc. Seems like that encapsulation should include this logic too. The `Descriptor` class is meant to be a more high level representation where functionality about high level concepts like apps, engines, configs core etc are implemented.

Apologies in advance if I am missing parts of the picture here, but still think maybe better to have it as part of `IODescriptor.base` - feels a bit more encapsulated that way - is the cyclic dependency on `Descriptor` that trips things up? Can we just do that import locally inside the method maybe?